### DISCUSSION TITLE
time series with no data in weekend or missing data

### DISCUSSION THREAD

**Discussion by LishuaiJing3** [alan-turing-institute/sktime#1419] [Q&A]
How to use sktime in the following two scenarios?
1) there is no data in weekend both in history and for forecast, how to use sktime for different algorithms?
2) What if there is missing data, for example holidays or perhaps other special events that you do not have the knowledge why the data is missing?

How to use sktime for these scenarios?

**Comment by fkiraly** [comment]
>there is no data in weekend both in history and for forecast, how to use sktime for different algorithms?

Just remove all the week-ends and use periods of five would be an option.
You may or may not want to change the index to "week days since [some fixed date]"

> What if there is missing data, for example holidays or perhaps other special events that you do not have the knowledge why the data is missing?

You could treat the missingness as an exogeneous indicator variable, and/or use a transformer or forecaster that can deal with missing data (tag `handles-missing-data`). You could also add in holiday/special event indicator variables.

**Comment by LishuaiJing3** [comment]
it seems that sktime can be smart enough to detect the 5 days pattern for finance data and only forecasting weekdays.
